sdk/core/System.Net.ClientModel/tests/client/MapsClient/MapsClient.cs

```diff
@@ -97,7 +92,7 @@ private ClientMessage CreateGetLocationRequest(string ipAddress, RequestOptions
 
         StringBuilder query = new();
         query.Append("api-version=");
-        query.Append(Uri.EscapeDataString(_apiVersion));
+        query.Append(Uri.EscapeDataString(options.PipelineOptions.ServiceVersion ?? LatestServiceVersion));
```

Comment:
This may be a lot of ask the generator to handle, and we may want to consider how we can store this in the cached pipeline when it's created.

Reply:
The current version addresses this by putting service version back on a client-specific options type derived from PipelineOptions.